mongomock/aggregate.py

```diff
@@ -984,6 +984,28 @@ def _find_matches_for_depth(query):
                 found_items.add(new_match['_id'])
         return new_matches
 
+    def _recursive_get(match, nested_fields):
+        first_field = nested_fields[0]
+        head = match.get(first_field)
+        if len(nested_fields) == 1:
+            # final/last field reached
+            yield head
+        else:
+            # more to go, get remaining from list/dict
+            remaining_fields = nested_fields[1:]
+            if isinstance(head, list):
+                for m in head:
+                    # yield from _recursive_get(m, remaining_fields)
+                    for answer in _recursive_get(m, remaining_fields):
+                        yield answer
+            elif isinstance(head, dict):
+                # yield from _recursive_get(head, remaining_fields)
+                for answer in _recursive_get(head, remaining_fields):
+                    yield answer
+            else:
+                # field doesn't exist, just let go
+                pass
```

Comment:
I'm not sure we need this code: dropping the else and pass statement should work the same.